include/aws/s3/private/s3_util.h

```diff
@@ -240,12 +249,21 @@ void aws_s3_add_user_agent_header(struct aws_allocator *allocator, struct aws_ht
  * object-size. All output arguments are optional.*/
 AWS_S3_API
 int aws_s3_parse_content_range_response_header(
-    struct aws_allocator *allocator,
     struct aws_http_headers *response_headers,
     uint64_t *out_range_start,
     uint64_t *out_range_end,
     uint64_t *out_object_size);
 
+/* Given a Content-Range header value as a byte cursor, parses the range-start, range-end and
+ * object-size. All output arguments are optional. This is an optimized version that doesn't
+ * require string allocation. */
+AWS_S3_API
```

Comment:
its in private header? should we just get rid of non optimized version?

Reply:
the previous version is taking the `struct aws_http_headers`. I already updated the previous function to skip the string allocation. I'll update the doc here to be less confusing.